crates/services/gas_price_service/src/ports.rs

```diff
@@ -43,12 +46,22 @@ pub trait GetDaSequenceNumber: Send + Sync {
     fn get_sequence_number(&self, block_height: &BlockHeight) -> Result<Option<u32>>;
 }
 
-pub trait TransactionableStorage: Send + Sync {
-    type Transaction<'a>
+pub trait TransactionableStorage
+where
+    Self: 'static,
+    Self: Send + Sync,
+    Self: GetMetadataStorage + GetDaSequenceNumber,
+{
+    type Transaction<'a>: AsUnrecordedBlocks
+        + SetMetadataStorage
+        + GetMetadataStorage
+        + SetDaSequenceNumber
+        + GetDaSequenceNumber
```

Comment:
Generally I think pushing many constraints into a single trait is an anti-pattern. I prefer to keep all the constraints as close to place that needs those constraints met. Namely, the `GasPriceService` needs these things to be transactionable together in order to construct the storage tx.

You can make the argument that the _domain_ requirement is that our storage is transactionable for these things. Which I would agree with, but then we need to include the `UnrecordedBlocks` as a requirement as well.

Reply:
I feel like I'd prefer these constraints on the RunableTask impl for GasPriceService than on some obfuscated trait, but if we can name this something informative, not just Transactionable then I might be okay with it. e.g. GasPriceServiceAtomicStorage or something like that :)

Reply:
Naming is fully up to you, I just tried to show you an exampel fo how it can looks like=)